3-UsingAPIs/Week2/homework/ex2-pokemonApp/index.js

```diff
@@ -1,35 +1,55 @@
-'use strict';
-/*------------------------------------------------------------------------------
-Complete the four functions provided in the starter `index.js` file:
-
-`fetchData`: In the `fetchData` function, make use of `fetch` and its Promise 
-  syntax in order to get the data from the public API. Errors (HTTP or network 
-  errors) should be logged to the console.
-
-`fetchAndPopulatePokemons`: Use `fetchData()` to load the pokemon data from the 
-  public API and populates the `<select>` element in the DOM.
-  
-`fetchImage`: Use `fetchData()` to fetch the selected image and update the 
-  `<img>` element in the DOM.
-
-`main`: The `main` function orchestrates the other functions. The `main` 
-  function should be executed when the window has finished loading.
-
-Try and avoid using global variables. Instead, use function parameters and 
-return values to pass data back and forth.
-------------------------------------------------------------------------------*/
-function fetchData(/* TODO parameter(s) go here */) {
+const buttonGetPokemon = document.createElement('button');
+document.body.appendChild(buttonGetPokemon);
+
+buttonGetPokemon.setAttribute('type', 'submit');
+buttonGetPokemon.textContent = 'Get Pokemon!';
+
+const form = document.createElement('form');
+document.body.appendChild(form);
+
+const selectPokemon = document.createElement('select');
+form.appendChild(selectPokemon);
+
+const pokemonImage = document.createElement('img');
+
+async function fetchData(url) {
   // TODO complete this function
+  const response = await fetch(url);
+  const data = await response.json();
+  return data.results;
 }
 
-function fetchAndPopulatePokemons(/* TODO parameter(s) go here */) {
+function fetchAndPopulatePokemons(element) {
   // TODO complete this function
+  element.forEach((pokemon) => {
+    const option = document.createElement('option');
+    selectPokemon.appendChild(option);
+    option.value = pokemon.name;
+    option.textContent = pokemon.name;
+  });
 }
 
-function fetchImage(/* TODO parameter(s) go here */) {
+selectPokemon.addEventListener('click', fetchImage);
+
+async function fetchImage() {
   // TODO complete this function
+  const urlOfThePokemonImages = `https://pokeapi.co/api/v2/pokemon/${selectPokemon.value}`;
```

Comment:
Do not construct this URL yourself because you are making assumptions on the format of this URL. Instead, use the `url` property of the pokemon data. This isolates you from problems if the owner of the API decides to make changes to the format of URL.